### PR TITLE
Added command line option to print page source on failure and enabled it for Travis

### PR DESCRIPTION
**Pull request by Dantemss**
To help debug sporadic failures

### CODE REVIEW

**Comment by Dantemss** [comment]
Conveniently, this failed immediately with a timeout. The page source makes it look like a legitimate timeout. What do you think @m1yag1? Should we increase the webview timeout?

**Comment by Dantemss** [comment]
The legacy error will be prevented by https://github.com/openstax/cnx-automation/pull/64 (I detect the site error on publish and retry the publish up to 3 times). Unless it fails 3 times in a row.

File: fixtures/base.py [hunk 1]
```diff
@@ -7,10 +7,18 @@
...
-def selenium(selenium):
+def selenium(request, selenium, pytestconfig):
     selenium.implicitly_wait(0)
-    return selenium
+    yield selenium
+    # request.node is an "item" because we use the default "function" scope
+    if pytestconfig.getoption('--print-page-source-on-failure') and \
+       request.node.rep_setup.passed and \
+       request.node.rep_call.failed:
+        print("\n------------------------------ Begin Page Source -------------------------------")
```
Comment (m1yag1) [inline fixtures/base.py]: single quotes please.

File: tests/conftest.py [hunk 2]
```diff
@@ -54,3 +58,16 @@ def pytest_addoption(parser):
...
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+
+    # set a report attribute for each phase of a call, which can
+    # be "setup", "call", "teardown"
+
+    setattr(item, "rep_" + rep.when, rep)
```
Comment (m1yag1) [inline tests/conftest.py]: string concatenation is generally frowned upon in python. You'll want to use `'rep_{}'.format(rep.when)`.

**Comment by Dantemss** [comment]
@m1yag1 changes made

**Comment by m1yag1** [comment]
Latest PR #80 has caused conflicts on this branch @Dantemss

**Pull request merged**